libs/category/src/reducers/category/category.reducer.ts

```diff
@@ -104,3 +116,96 @@ export function categoryReducer(state = initialState, action: DaffCategoryAction
       return state;
   }
 }
+
+function toggleCategoryFilter(
+	toggledFilter: DaffToggledCategoryFilterRequest, 
+	appliedFilters: DaffCategoryFilterRequest[]
+): DaffCategoryFilterRequest[] {
+	return isFilterApplied(toggledFilter, appliedFilters)
+		? removeFilter(toggledFilter, appliedFilters)
+		: addFilter(toggledFilter, appliedFilters)
+}
+
+function isFilterApplied(toggledFilter: DaffToggledCategoryFilterRequest, appliedFilters: DaffCategoryFilterRequest[]): boolean {
+	const filterMatchingName = appliedFilters.find(filter => toggledFilter.name === filter.name);
```

Comment:
I don't think making subfunctions changes anything about the single responsibility of the parent function, because it still ultimately does those things. But I do think it makes it more readable. Changed